### PR TITLE
feat: initial setup for submitting article to squad

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Introduce generic way of submitting article to a squad

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-851 #done

### CODE REVIEW

**Review by sshanzel — COMMENTED** [review]
I think this is the best way to go as the component was already structured to handle different types through `switch-case` as you did.

**Comment by el** [comment]
https://user-images.githubusercontent.com/1488164/213143749-ab8e0c4b-2e31-4b94-803c-83079ba6a983.mov

File: packages/shared/src/components/modals/PostToSquadModal.tsx [hunk 1]
```diff
@@ -1,53 +1,86 @@
...
+    onRequestClose,
+  };
 
   return (
     <Modal
       isOpen={isOpen}
       kind={Modal.Kind.FixedCenter}
       size={Modal.Size.Small}
       onRequestClose={onRequestClose}
+      steps={post ? undefined : modalSteps}
```
[the remark this answers is not in the document]
Comment (sshanzel) [reply packages/shared/src/components/modals/PostToSquadModal.tsx:73]: I'd prefer `undefined`, personally.

File: packages/shared/src/hooks/useFeed.ts [hunk 2]
```diff
@@ -25,9 +25,10 @@ export type PostItem = {
   page: number;
   index: number;
 };
+export type NewSquadPostItemm = { type: 'new_squad_post'; action: () => void };
```
Comment (sshanzel) [inline packages/shared/src/hooks/useFeed.ts:28]: Just a typo.
```suggestion
export type NewSquadPostItem = { type: 'new_squad_post'; action: () => void };
```

**Review by sshanzel — APPROVED** [review]
Looks amazing to me! Just that one typo before merging 🚢

File: packages/shared/src/hooks/useFeed.ts (hunk 2 above)
Comment (rebelchris) [reply packages/shared/src/hooks/useFeed.ts:28]: Sharp eyes 🦅

**Pull request merged**